pycket/pycket-lang/zoTransform.rkt

```diff
@@ -0,0 +1,581 @@
+           (hash* 'numerator (handle-number num)
+                  'denominator (handle-number den))))
+         ]
+    [else ; this part assumes (for the moment) it's a complex num
+     (let ([real (real-part racket-num)]
+           [imag (imag-part racket-num)])
+       (hash* 'real-part (handle-number real)
+              'imag-part (handle-number imag)))]))
+
+(define (handle-boolean racket-bool)
+  (if racket-bool
+      (hash* 'quote true)
+      (hash* 'quote false)))
+
+(define (handle-string racket-str)
+  (hash* 'quote (hash* 'string racket-str)))
+
+(define (handle-symbol racket-sym)
+  (hash* 'quote (hash* 'toplevel (symbol->string racket-sym))))
+
+(define (get-primval-name id)
+  (symbol->string (hash-ref primitive-table id)))
+  
+(define (handle-primval operation toplevels)
+  (let* ((id (primval-id operation))
+         (operator-name (get-primval-name id)))
+    (hash* 'source-name operator-name)))
+
+(define (handle-application app-form toplevels localref-stack)
+  (let* ((rator (application-rator app-form))
+         (rands (application-rands app-form))
+         (newlocalstack (append (map (lambda (x) 'app-empty-slot) (range (length rands)))
+                                localref-stack))
+         ;; the application pushes empty slots to run body over, so it will push the current local references further
+         ;; we kinda simulate it here to make the localref pos indices point to the right identifier
+         (operands-evaluated (map (λ (rand) (to-ast-single rand toplevels newlocalstack)) rands))
+         )
+    (hash* 'operator (to-ast-single rator toplevels localref-stack)
+           'operands operands-evaluated)))
+
+(define (handle-lambda lam-form toplevels localref-stack is-inlined)
+  (let* ((name (lam-name lam-form))
+         (source (if (null? name) '()
+                     (if (vector? name)
+                         (hash* '%p (path->string (vector-ref name 1)))
+                         (if (not (symbol? name)) (error 'handle-lambda "we have a non symbol/vector name in a lam form")
+                             (let* ((collects-dir (path->string (find-collects-dir)))
+                                    (usual-prefix "/racket/private/") ; TODO: figure out why they have a new way of naming lam's
+                                    (lamname (if (not (string-contains? (symbol->string name) ".../more-scheme.rkt"))
+                                                 (error 'handle-lambda "lam name has an unusual form")
+                                                 "more-scheme.rkt")))
+                               (hash* '%p (string-append collects-dir usual-prefix lamname)))))))
+         (position 321) ;; don't know what exactly are these two
+         (span 123) 
+         ;; module seems to be the same for every lambda form,
+         ;; pointing to a private module about chaperones/impersonators
+         (module (hash* '%mpi (hash* '%p (string-append collectsDir "/racket/private/kw.rkt"))))
```

Comment:
Yeah actually the pycket tests are revealing all kinds of special cases like this one that I overlooked before. Working on it.